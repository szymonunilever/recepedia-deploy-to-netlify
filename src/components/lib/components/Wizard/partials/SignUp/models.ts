export interface SignUpProps {
  title: string;
  formContent: AppContent.GeneratedForm.Content;
  actionCallback: () => void;
  containerClass: string;
  stepId: string;
}
