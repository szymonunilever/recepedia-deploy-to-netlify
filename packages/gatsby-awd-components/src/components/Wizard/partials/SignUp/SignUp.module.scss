/////----------- CUSTOM -----------/////
.wizard__buttons {
  box-sizing: border-box;
}
:global {
  .wizard--form {
    .wizard {
      &__container {
        height: auto;
        padding-bottom: 20px;
      }
    }
  }
}
