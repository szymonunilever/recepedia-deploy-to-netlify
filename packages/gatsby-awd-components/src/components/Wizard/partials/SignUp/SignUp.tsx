import React, { Fragment, FunctionComponent } from 'react';
import { SignUpProps } from './models';
import { TagName, Text } from '../../../Text';
import GeneratedForm from '../../../GeneratedForm';
import { Button } from '../../../Button';
import theme from './SignUp.module.scss';
import cx from 'classnames';

const SignUp: FunctionComponent<SignUpProps> = ({
  actionCallback,
  formContent,
  title,
}) => (
  <Fragment>
    <div className="wizard__info">
      <div className="wizard__title">
        <Text className="" tag={TagName.h2} text={title} />
      </div>
    </div>
    <GeneratedForm
      onSubmit={actionCallback}
      content={formContent}
      className=""
      shouldValidate
    />
    <div className={cx(theme.wizard__buttons, 'wizard__buttons')}>
      <Button
        className="wizard__button wizard__button--secondary"
        onClick={actionCallback}
        content={{ label: 'Skip' }}
      />
    </div>
  </Fragment>
);

export default SignUp;
