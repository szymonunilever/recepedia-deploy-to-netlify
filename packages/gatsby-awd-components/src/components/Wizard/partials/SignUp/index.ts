export { default } from './SignUp';
export { default as SignUp } from './SignUp';
export * from './models';
