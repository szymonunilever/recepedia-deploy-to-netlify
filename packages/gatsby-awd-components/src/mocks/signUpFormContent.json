{
  "title": "Easy & healthy recipes every week",
  "fields": [
    {
      "name": "email input",
      "type": "email",
      "placeholder": "Your email address",
      "validationRules": [
        {
          "type": "required",
          "errorMessage": "This field is required."
        },
        {
          "type": "maxLength",
          "value": 20,
          "errorMessage": "This field is too long."
        },
        {
          "type": "minLength",
          "value": 4,
          "errorMessage": "This field is too short."
        },
        {
          "type": "email",
          "errorMessage": "Should be a valid email."
        }
      ]
    }
  ],
  "submitButton": {
    "label": "Sign Up!"
  }
}
