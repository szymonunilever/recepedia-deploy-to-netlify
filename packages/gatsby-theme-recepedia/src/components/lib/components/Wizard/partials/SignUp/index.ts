export { default } from './SignUp';
export { default as SignUp } from './SignUp';
