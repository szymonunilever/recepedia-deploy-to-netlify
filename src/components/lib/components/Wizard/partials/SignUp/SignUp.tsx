import React, { Fragment, FunctionComponent } from 'react';
import { SignUpProps } from './models';
import { TagName, Text } from '../../../Text';
import GeneratedForm from '../../../GeneratedForm';
import { Button } from '../../../common/Button';

const SignUp: FunctionComponent<SignUpProps> = ({
  actionCallback,
  formContent,
  title,
}) => (
  <Fragment>
    <div className="wizard__info">
      <div className="wizard__title">
        <Text className="" tag={TagName.h2} text={title} />
      </div>
    </div>
    <GeneratedForm
      onSubmit={actionCallback}
      content={formContent}
      className=""
      shouldValidate
    />
    <div className="wizard__buttons">
      <Button
        className="wizard__button wizard__button--secondary"
        onClick={actionCallback}
        content={{ label: 'Skip' }}
      />
    </div>
  </Fragment>
);

export default SignUp;
