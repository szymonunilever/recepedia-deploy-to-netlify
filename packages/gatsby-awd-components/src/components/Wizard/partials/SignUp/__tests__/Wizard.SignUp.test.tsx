import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import SignUp, { SignUpProps } from '../index';
import toJson from 'enzyme-to-json';
import { formContentMock } from 'src/mocks/signUpFormContent.js';

describe('<SignUp />', () => {
  let wrapper: ReactWrapper;
  const signUpProps: SignUpProps = {
    actionCallback: () => console.log(),
    formContent: formContentMock,
    title: 'Lorem ipsum',
    containerClass: 'test',
    stepId: 'test',
  };

  beforeEach(() => {
    wrapper = mount(<SignUp {...signUpProps} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
