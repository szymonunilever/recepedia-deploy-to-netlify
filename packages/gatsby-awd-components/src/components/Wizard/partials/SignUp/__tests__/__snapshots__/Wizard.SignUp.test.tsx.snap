// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> matches the snapshot 1`] = `
<SignUp
  actionCallback={[Function]}
  containerClass="test"
  formContent={
    Object {
      "fields": Array [
        Object {
          "name": "email input",
          "placeholder": "Your email address",
          "type": "email",
          "validationRules": Array [
            Object {
              "errorMessage": "This field is required.",
              "type": "required",
            },
            Object {
              "errorMessage": "This field is too long.",
              "type": "maxLength",
              "value": 20,
            },
            Object {
              "errorMessage": "This field is too short.",
              "type": "minLength",
              "value": 4,
            },
            Object {
              "errorMessage": "Should be a valid email.",
              "type": "email",
            },
          ],
        },
      ],
      "submitButton": Object {
        "label": "Sign Up!",
      },
      "title": "Easy & healthy recipes every week",
    }
  }
  stepId="test"
  title="Lorem ipsum"
>
  <div
    className="wizard__info"
  >
    <div
      className="wizard__title"
    >
      <Text
        className=""
        tag={1}
        text="Lorem ipsum"
      >
        <h2
          className=""
        >
          Lorem ipsum
        </h2>
      </Text>
    </div>
  </div>
  <GeneratedForm
    className=""
    content={
      Object {
        "fields": Array [
          Object {
            "name": "email input",
            "placeholder": "Your email address",
            "type": "email",
            "validationRules": Array [
              Object {
                "errorMessage": "This field is required.",
                "type": "required",
              },
              Object {
                "errorMessage": "This field is too long.",
                "type": "maxLength",
                "value": 20,
              },
              Object {
                "errorMessage": "This field is too short.",
                "type": "minLength",
                "value": 4,
              },
              Object {
                "errorMessage": "Should be a valid email.",
                "type": "email",
              },
            ],
          },
        ],
        "submitButton": Object {
          "label": "Sign Up!",
        },
        "title": "Easy & healthy recipes every week",
      }
    }
    onSubmit={[Function]}
    shouldValidate={true}
  >
    <GeneratedFormInstance
      className=""
      content={
        Object {
          "fields": Array [
            Object {
              "name": "email input",
              "placeholder": "Your email address",
              "type": "email",
              "validationRules": Array [
                Object {
                  "errorMessage": "This field is required.",
                  "type": "required",
                },
                Object {
                  "errorMessage": "This field is too long.",
                  "type": "maxLength",
                  "value": 20,
                },
                Object {
                  "errorMessage": "This field is too short.",
                  "type": "minLength",
                  "value": 4,
                },
                Object {
                  "errorMessage": "Should be a valid email.",
                  "type": "email",
                },
              ],
            },
          ],
          "submitButton": Object {
            "label": "Sign Up!",
          },
          "title": "Easy & healthy recipes every week",
        }
      }
      hasCaptcha={false}
      onSubmit={[Function]}
      shouldValidate={true}
    >
      <Form
        className="generatedForm generated-form"
        onSubmit={[Function]}
        render={[Function]}
        subscription={
          Object {
            "submitting": true,
          }
        }
      >
        <ReactFinalForm
          className="generatedForm generated-form"
          onSubmit={[Function]}
          render={[Function]}
          subscription={
            Object {
              "submitting": true,
            }
          }
        >
          <form
            className="generatedForm generated-form"
            noValidate={true}
            onSubmit={[Function]}
          >
            <div
              className="generated-form__container"
            >
              <Text
                className="generated-form__title"
                tag={6}
                text="Easy & healthy recipes every week"
              >
                <div
                  className="generated-form__title"
                >
                  Easy & healthy recipes every week
                </div>
              </Text>
              <div
                className="generated-form__fields"
              >
                <GeneratedField
                  className="generated-form__item"
                  content={
                    Object {
                      "name": "email input",
                      "placeholder": "Your email address",
                      "type": "email",
                      "validationRules": Array [
                        Object {
                          "errorMessage": "This field is required.",
                          "type": "required",
                        },
                        Object {
                          "errorMessage": "This field is too long.",
                          "type": "maxLength",
                          "value": 20,
                        },
                        Object {
                          "errorMessage": "This field is too short.",
                          "type": "minLength",
                          "value": 4,
                        },
                        Object {
                          "errorMessage": "Should be a valid email.",
                          "type": "email",
                        },
                      ],
                    }
                  }
                  key="0"
                  shouldValidate={true}
                >
                  <Input
                    content={
                      Object {
                        "placeholder": "Your email address",
                      }
                    }
                    error={[Function]}
                    name="email input"
                    rules={
                      Array [
                        Object {
                          "errorMessage": "This field is required.",
                          "type": "required",
                        },
                        Object {
                          "errorMessage": "This field is too long.",
                          "type": "maxLength",
                          "value": 20,
                        },
                        Object {
                          "errorMessage": "This field is too short.",
                          "type": "minLength",
                          "value": 4,
                        },
                        Object {
                          "errorMessage": "Should be a valid email.",
                          "type": "email",
                        },
                      ]
                    }
                    type="email"
                    validate={[Function]}
                  >
                    <Field
                      name="email input"
                      validate={[Function]}
                    >
                      <Field
                        name="email input"
                        validate={[Function]}
                      >
                        <div
                          className="field"
                        >
                          <label
                            className="field__label"
                          >
                            <span
                              className="field__label-text"
                            />
                            <div
                              className="field__wrap"
                            >
                              <input
                                className="field__input"
                                name="email input"
                                onBlur={[Function]}
                                onChange={[Function]}
                                onFocus={[Function]}
                                placeholder="Your email address"
                                type="email"
                                value=""
                              />
                            </div>
                          </label>
                          <p
                            className="field__hint"
                          />
                        </div>
                      </Field>
                    </Field>
                  </Input>
                </GeneratedField>
              </div>
              <div
                className="buttons"
              >
                <Component
                  className="generated-form__button--primary"
                  isDisabled={false}
                  type="submit"
                >
                  <button
                    className="button button generated-form__button--primary"
                    data-componentname="button"
                    data-componentview="button"
                    disabled={false}
                    hidden={false}
                    onClick={[Function]}
                    type="submit"
                  >
                    Sign Up!
                  </button>
                </Component>
              </div>
            </div>
          </form>
        </ReactFinalForm>
      </Form>
    </GeneratedFormInstance>
  </GeneratedForm>
  <div
    className="wizard__buttons wizard__buttons"
  >
    <Component
      className="wizard__button wizard__button--secondary"
      content={
        Object {
          "label": "Skip",
        }
      }
      onClick={[Function]}
    >
      <button
        className="button button wizard__button wizard__button--secondary"
        data-componentname="button"
        data-componentview="button"
        disabled={false}
        hidden={false}
        onClick={[Function]}
        type="button"
      >
        <span
          className="button__label"
        >
          Skip
        </span>
      </button>
    </Component>
  </div>
</SignUp>
`;
